### ISSUE TITLE
Field annotation for method

### ISSUE THREAD

**Issue by msimple** [sdaschner/jaxrs-analyzer#131]
Hi!
We have a project where we use class builders on abstract classes and instead of fields we have to create methods. I've been trying to find a way to use some annotation that would show a non existing field in the generated json but I couldn't find any. Did I miss something or this is not supported?

Example code segment:
public abstract class ExampleClass{
	@JsonProperty public abstract String field1();
	@JsonProperty public abstract String field2();
}
Desired swagger definition:
ExampleClass:
    properties:
      field1:
        type: integer
      field2:
        type: string

**Comment by sdaschner** [comment]
Hi there! Your @JsonProperty annotation comes from Jackson, right? I'm quite sure that not full Jackson scenarios are supported -- especially for inheritance...

The plan is to support the JSON-B standard in the future... So that would have priority.
However, we're more than happy to receive PRs and contributions!

**Comment by msimple** [comment]
Yes, it is Jackson related, we are using immutable class generation. If there is another annotation that recognizes methods as fields we would be happy to use that as well.

I tried to look into it in the source code, but I got stuck at the bytecode level. Maybe my approach was bad, but I couldn't quite follow the core of the code. Because of this I can't really tell how much handholding and time I would need to be able to contribute. Our company has no resources for this right now so I can only offer my free time.
